src/python/genny/csv2.py

```diff
@@ -82,42 +82,43 @@ def __iter__(self):
     def __next__(self):
         return self._parse_into_intermediate_csv(next(self.raw_reader))
 
-    def _parse_into_intermediate_csv(self, line):
-        for i in range(len(line)):
-
-            # Strip spaces from lines
-            line[i] = line[i].strip()
+    def _parse_into_intermediate_csv(self, csv_line):
+        op = csv_line[_OpColumns.OPERATION].rstrip()
+        actor = csv_line[_OpColumns.ACTOR].rstrip()
+        # Convert timestamp from ns to ms and add offset.
+        unix_time = (int(csv_line[_OpColumns.TIMESTAMP]) + self.unix_time_offset) / (1000 * 1000)
 
-            # Convert string digits to ints
-            if line[i].isdigit():
-                line[i] = int(line[i])
+        # Transform output into IntermediateCSV format.
+        # Python list assignment too slow so we construct the output array directly.
```

Comment:
30 nanoseconds per list element assignment doesn't seem that slow to me. I think the problem comes down to how the list is being constructed via a list comprehension.

If the number of columns isn't expected to change at runtime, then we should we using tuples over lists.

```
$ python3 -m timeit 'a = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]'
$ python3 -m timeit 'a = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]'
5000000 loops, best of 5: 96.8 nsec per loop

$ python3 -m timeit 'a = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]; a[0] = 0; a[1] = 1; a[2] = 2; a[3] = 3; a[4] = 4; a[5] = 5; a[6] = 6; a[7] = 7; a[8] = 8; a[9] = 9; a[10] = 10'
1000000 loops, best of 5: 401 nsec per loop

$ python3 -m timeit 'a = [None for _ in range(len(["unix_time", "ts", "thread", "operation", "duration", "outcome", "n", "ops", "errors", "size", "workers"]))]; a[0] = 0; a[1] = 1; a[2] = 2; a[3] = 3; a[4] = 4; a[5] = 5; a[6] = 6; a[7] = 7; a[8] = 8; a[9] = 9; a[10] = 10'
200000 loops, best of 5: 1.41 usec per loop

$ python3 -m timeit 'a = [None for _ in range(len(["unix_time", "ts", "thread", "operation", "duration", "outcome", "n", "ops", "errors", "size", "workers"]))]'
200000 loops, best of 5: 1.05 usec per loop

$ python3 -m timeit 'a = [None] * 11'
2000000 loops, best of 5: 149 nsec per loop

$ python3 -m timeit 'a = [None] * len(["unix_time", "ts", "thread", "operation", "duration", "outcome", "n", "ops", "errors", "size", "workers"])'
1000000 loops, best of 5: 297 nsec per loop

$ python3 -m timeit 'a = [None] * len(("unix_time", "ts", "thread", "operation", "duration", "outcome", "n", "ops", "errors", "size", "workers"))'
1000000 loops, best of 5: 228 nsec per loop
```

Reply:
Thanks for the tip! I removed list comprehension in favor of multiplication